scripts/e2e-all-features.sh

```
@@ -85,6 +85,25 @@ else no "conversation read" "no claude session found"; fi
 [ "$(hcode "${A[@]}" "${J[@]}" -X POST -d '{"sessionId":"../etc","cwd":"/tmp"}' "${BASE}/api/sessions/connect")" = 400 ] && ok "/api/sessions/connect rejects path-traversal id (400)" || no "connect id guard" "no 400"
 [ "$(hcode "${A[@]}" "${J[@]}" -X POST -d '{"sessionId":"abc","cwd":"/etc"}' "${BASE}/api/sessions/connect")" = 400 ] && ok "/api/sessions/connect rejects cwd outside home (400)" || no "connect cwd guard" "no 400"
 
+echo ""; echo "── 5b. Session-Over-Sessions (SoS coordinator) ──"
+SS=$(curl -s "${A[@]}" "${J[@]}" -X POST -d '{"label":"e2e-sos"}' "${BASE}/api/sos")
+SSID=$(echo "$SS" | jget "d['id']")
+[ -n "$SSID" ] && ok "POST /api/sos creates super-session" || no "create super-session" "no id ($SS)"
+curl -s "${A[@]}" "${BASE}/api/sos" | grep -q "$SSID" && ok "GET /api/sos lists it" || no "list super-sessions" "missing"
+CJ=$(find "$HOME/.claude/projects" -name "*.jsonl" -size +2k 2>/dev/null | head -1)
+if [ -n "$CJ" ]; then
+  MADD=$(hcode "${A[@]}" "${J[@]}" -X POST -d "{\"sessionKey\":\"${CJ}\",\"provider\":\"claude-code\"}" "${BASE}/api/sos/${SSID}/members")
+  [ "$MADD" = 201 ] && ok "POST /api/sos/[id]/members adds a member (201)" || no "add member" "got $MADD"
```

Shellcheck (0.11.0)

[info] Note that A && B || C is not if-then-else. C may run when A is true.

(SC2015)